crates/holochain/src/conductor/conductor.rs

```diff
@@ -837,12 +838,107 @@ mod network_impls {
 
         pub(crate) async fn network_info(
             &self,
-            dnas: &[DnaHash],
+            payload: &NetworkInfoRequestPayload,
         ) -> ConductorResult<Vec<NetworkInfo>> {
+            use holochain_sqlite::sql::sql_cell::SUM_OF_RECEIVED_BYTES_SINCE_TIMESTAMP;
+
+            let NetworkInfoRequestPayload {
+                agent_pub_key,
+                dnas,
+                last_time_queried,
+            } = payload;
+
             futures::future::join_all(dnas.iter().map(|dna| async move {
-                let d = self.holochain_p2p.get_diagnostics(dna.clone()).await?;
-                let fetch_pool_info = d.fetch_pool.info([dna.to_kitsune()].into_iter().collect());
-                ConductorResult::Ok(NetworkInfo { fetch_pool_info })
+                let diagnostics = self.holochain_p2p.get_diagnostics(dna.clone()).await?;
+                let fetch_pool_info = diagnostics
+                    .fetch_pool
+                    .info([dna.to_kitsune()].into_iter().collect());
+
+                // query number agents from peer db
+                let db = { self.p2p_agents_db(dna) };
+                let permit = db.conn_permit().await;
+                let mut conn = db.with_permit(permit)?;
+                let agent_infos = conn.p2p_list_agents()?;
+                let number_of_peers = agent_infos.len();
+
+                // query arc size and extrapolated coverage and estimate total peers
+                let cell_id = CellId::new(dna.as_hash().clone(), agent_pub_key.clone());
+                let agent_infos = self
+                    .get_agent_infos(Some(cell_id.clone()))
+                    .await
+                    .map_err(|_| ConductorError::CellMissing(cell_id.clone()))?;
+                let arc_size = agent_infos[0].storage_arc.coverage();
+                let extrapolated_coverage =
+                    conn.p2p_extrapolated_coverage(agent_infos[0].storage_arc.inner().into())?;
+                let total_peers = (extrapolated_coverage[0] / arc_size) as u64;
```

Comment:
This may be wrong as I'm guessing what the extrapolated coverage represents.

Reply:
The indirection through the arc-size certainly seems like it would lessen the accuracy of this extrapolation, since peers are welcome to set their arc-sized to zero. If this metric is really intended to literally just be a guess at the peer count, a more direct extrapolation would be to just get the raw peer count within our storage arc, then divide out the factor of our storage arc size relative to the whole.

Obviously this won't work if our *own* storage arc is zero.